mps-model-adapters/src/main/kotlin/org/modelix/model/mpsadapters/MPSNode.kt

```diff
@@ -62,20 +60,25 @@ data class MPSNode(val node: SNode) : IDefaultNodeAdapter, IReplaceableNode {
         }
 
     override fun replaceNode(concept: ConceptReference?): INode {
-        requireNotNull(concept) { "Can't replace $node with null concept. Use BaseConcept explicitly." }
+        val parent = requireNotNull(node.parent) { "Cannot replace node `$node` because it has no parent." }
+
+        requireNotNull(concept) { "Cannot replace node `$node` with a null concept. Explicitly specify a concept (e.g., `BaseConcept`)." }
```

Comment:
Could be moved to the start (paragraph) of the method as this is a precondition.

Reply:
Done